ios/app/watchos/extension/ExtensionDelegate.swift

```diff
@@ -0,0 +1,112 @@
+                // Be sure to complete the background task once you’re done.
+                backgroundTask.setTaskCompletedWithSnapshot(false)
+            case let snapshotTask as WKSnapshotRefreshBackgroundTask:
+                // Snapshot tasks have a unique completion call, make sure to set your expiration date
+                snapshotTask.setTaskCompleted(restoredDefaultState: true, estimatedSnapshotExpiration: Date.distantFuture, userInfo: nil)
+            case let connectivityTask as WKWatchConnectivityRefreshBackgroundTask:
+                // Be sure to complete the connectivity task once you’re done.
+                connectivityTask.setTaskCompletedWithSnapshot(false)
+            case let urlSessionTask as WKURLSessionRefreshBackgroundTask:
+                // Be sure to complete the URL session task once you’re done.
+                urlSessionTask.setTaskCompletedWithSnapshot(false)
+            default:
+                // make sure to complete unhandled task types
+                task.setTaskCompletedWithSnapshot(false)
+            }
+        }
+    }
+
+    func session(_ session: WCSession, activationDidCompleteWith
+        activationState: WCSessionActivationState, error: Error?) {
+        if let error = error {
+            print("WATCH Session activation failed with error: \(error.localizedDescription)")
+            return
+        }
+        print("WATCH Session activated with state: \(activationState.rawValue)")
+    }
+
+    func session(_ session: WCSession, didReceiveApplicationContext applicationContext: [String : Any]) {
+        let newContext = JitsiMeetContext(context: applicationContext)
+
+        print("WATCH got new context: \(newContext.description)");
+      
+        // Update context on the root controller which displays the recent list
+        let controller = WKExtension.shared().rootInterfaceController as! InterfaceController
+        controller.updateUI(newContext)
+
+        // If the current controller is not the in-call controller and we have a
+        // conference URL, show the in-call controller
+        if let currentController = WKExtension.shared().visibleInterfaceController as? InterfaceController {
+            // Go to the in-call controler only if the conference URL has changed, because the user may have clicked the back button
+            if newContext.conferenceURL != "NULL" && currentContext.conferenceURL != newContext.conferenceURL {
+                DispatchQueue.main.async {
+                    currentController.pushController(withName: "InCallController", context: newContext)
+                }
+            }
+        } else if let inCallController = WKExtension.shared().visibleInterfaceController as? InCallController {
+            if newContext.conferenceURL == "NULL" {
+                DispatchQueue.main.async {
+                    inCallController.popToRootController()
+                }
+            } else {
+                inCallController.updateUI(newContext)
```

Comment:
this should probably also run in the main queue. actually, probably the whole function should, WDYT?